### DISCUSSION TITLE
Will the 4.x version of Freeform be supported for a period of time?

### DISCUSSION THREAD

**Discussion by croxton** [solspace/craft-freeform#1031] [Questions & Answers] [answered]
I have a few sites close to launch soon and I'm reluctant to upgrade Freeform to 5.x while it is still relatively new and in the 'teething problems' stage. The upgrade to 5 requires quite significant changes, as well as the need to recreate CRM OAuth2 tokens (something that it took weeks to arrange with one client, for example). Therefore, I'd rather wait for Craft 5 and upgrade Freeform to 5.x at the same time.

However, my concern is around security and bugs: will the 4.x branch be patched if there is a security issue in the next, say, 1 year? How long will you endeavour to provide security patches and bug fixes for it? 

Many thanks.

**Comment by kjmartens** [comment, marked as answer]
Hi @croxton,

That's a very good plan.

We will continue to support Freeform 4 for the foreseeable future. Currently, we will still address any security vulnerabilities and critical issues on Freeform 3.x. There will be a few years of that available for Freeform 4.x 🙂

**Comment by croxton** [comment]
That's excellent to know! Thank you.

**Reply by kjmartens** [reply]
You're welcome! I will just reopen this so it's easier for others to discover it if they have the same concerns. 🙂
